plist_types.go

```diff
@@ -46,6 +46,47 @@ func (p *cfDictionary) sort() {
 	sort.Sort(p)
 }
 
+func (c *cfDictionary) filterNodes(nodes []Node) (keys []string, values []cfValue, filtered []Node) {
+	m := c.toMap()
+	add := map[string]bool{}
+	for _, node := range nodes {
+		key := node.Value()
+		if _, ok := node.(*Annotation); ok {
+			keys = append(keys, key)
+			values = append(values, nil)
+			add[key] = true
```

Comment:
Are Annotations guaranteed to be unique at the same level of the tree? Not sure if you can have the same comment appear twice in the same struct, or it that only happens with malformed plists.